features/step_definitions/stepdefs.rb

```diff
@@ -0,0 +1,7 @@
+Then('it has {int} error/errors and {int} warning/warnings') do |error, warning|
```

Comment:
to follow cucumber style convention, we should use conditional here and use the point of view of the user (*I run*, ...). I also think the `foo/bar` syntax is less suited than `foo(s)` here:

```suggestion
Then('I should see {int} error(s) and {int} warning(s)') do |error, warning|
```

Reply:
I looked at the style, but first tried regexes. That didn't work, but this looks cleaner.

Reply:
Hehe, I should probably have quoted "style" because there is no real reference / ultimate source of truth that I am aware of: much more experience that piles up with the years :shrug: (even if I only do some cucumber 3 of 4 times a year).

I don't remember which version of gherkin changed the game and allowed this flexible syntax instead of bare regular expressions.

Anyway, the doc points to this repository that is handy:
https://github.com/cucumber/cucumber-expressions